chat-sdk-ui/src/main/java/co/chatsdk/ui/threads/ThreadEditDetailsActivity.java

```diff
@@ -0,0 +1,162 @@
+        super.onCreate(savedInstanceState);
+
+        threadEntityID = getIntent().getStringExtra(InterfaceManager.THREAD_ENTITY_ID);
+        if (threadEntityID != null) {
+            thread = StorageManager.shared().fetchThreadWithEntityID(threadEntityID);
+        }
+
+        setContentView(R.layout.chat_sdk_activity_edit_thread_details);
+        initViews();
+    }
+
+    protected void initViews() {
+        actionBar = getSupportActionBar();
+        actionBar.setHomeButtonEnabled(true);
+
+        nameInput = findViewById(R.id.chat_sdk_edit_thread_name_et);
+        saveButton = findViewById(R.id.chat_sdk_edit_thread_update_b);
+
+        if (thread != null) {
+            actionBar.setTitle(Strings.nameForThread(thread));
+            nameInput.setText(thread.getName());
+            saveButton.setText(R.string.update_thread);
+
+            // TODO: permanently move thread name into meta data
+            ThreadMetaValue nameMetaValue = thread.metaValueForKey("name");
+            if (nameMetaValue != null)
+                nameInput.setText(nameMetaValue.getValue());
+        } else {
+            saveButton.setEnabled(false);
+        }
+
+        nameInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                saveButton.setEnabled(!nameInput.getText().toString().isEmpty());
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
+        });
+
+        saveButton.setOnClickListener(v -> {
+            didClickOnSaveButton();
+        });
+
+    }
+
+    protected void didClickOnSaveButton() {
+        final String threadName = nameInput.getText().toString();
+        if (thread == null) {
+            showOrUpdateProgressDialog(getString(R.string.add_public_chat_dialog_progress_message));
+
+            disposableList.add(ChatSDK.publicThread().createPublicThreadWithName(threadName)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe((newThread, throwable) -> {
+                        if (throwable == null) {
+                            // TODO: permanently move thread name into meta data
+                            newThread.setMetaValue("name", threadName);
+                            disposableList.add(ChatSDK.thread().pushThreadMeta(newThread).subscribe(() -> {
+                                dismissProgressDialog();
+                                ToastHelper.show(ChatSDK.shared().context(), String.format(getString(R.string.thread__created), threadName));
+
+                                finish();
+                                ChatSDK.ui().startChatActivityForID(ChatSDK.shared().context(), newThread.getEntityID());
```

Comment:
Why is finish called before ChatSDK.ui call?

Reply:
Because in this case the `ThreadEditDetailsActivity` is used to create a new thread and the user should not be able to go back to that activity when pressing the back button

Reply:
Could you add a comment to explain this?